formatTest/unit_tests/expected_output/infix.re

```diff
@@ -348,3 +368,586 @@ let myFunc
+      somethingElse :
+      (arr.(0) = somethingElse) ? x : z;
+    (
+      something ?
+        somethingElse : arr.(0) = somethingElse
+    ) ?
+      x : z;
+
+    /** These should be parsed the same */
+    something ?
+      somethingElse :
+      bigArr.{0} = somethingElse ? x : bigArr.{0};
+    something ?
+      somethingElse :
+      bigArr.{0} = somethingElse ? x : bigArr.{0};
+    /* Not: */
+    something ?
+      somethingElse :
+      (bigArr.{0} = somethingElse) ? x : z;
+    (
+      something ?
+        somethingElse :
+        bigArr.{0} = somethingElse
+    ) ?
+      x : z;
+
+    /** These should be parsed the same */
+    something ?
+      somethingElse :
+      arr.[0] = somethingElse ? x : arr.[0];
+    something ?
+      somethingElse :
+      arr.[0] = somethingElse ? x : arr.[0];
+    /* Not: */
+    something ?
+      somethingElse :
+      (str.[0] = somethingElse) ? x : z;
+    (
+      something ?
+        somethingElse : str.[0] = somethingElse
+    ) ?
+      x : z;
+
+    /**
+     * It creates a totally different meaning when parens group the :
+     */
+    x.contents =
+      something ?
+        (x.contents = somethingElse: x) : z;
+    y = something ? (y = somethingElse: x) : z;
+    arr.(0) =
+      something ?
+        (arr.(0) = somethingElse: x) : z;
+    bigArr.{0} =
+      something ?
+        (bigArr.{0} = somethingElse: x) : z;
+    str.[0] =
+      something ?
+        (str.[0] = somethingElse: x) : z;
+
+    /**
+     * Various precedence groupings.
+     */
+    true ? true ? false : false : false;
+    /* Is the same as */
+    true ? true ? false : false : false;
+    /*
+     * Just some examples of how prefix will be printed.
+     */
+    - x + (something.contents = y);
+    - x + (something = y);
+    - x + something.contents := y;
+    - x + something := y;
+    x + (- (something.contents = y));
+    x + (- (something = y));
+    x + (- something.contents) := y;
+    x + (- something) := y;
+    x.contents || something + 1 ?
+      - hello : goodbye;
+    bigArr.{0} || - something + 1 ?
+      hello : goodbye;
+    let result = - x + (something.contents = y);
+    /* Prefix minus is actually sugar for regular function identifier ~-*/
+    let result = 2 + (- add 4 0);
+    /* Same as */
+    let result = 2 + (- add) 4 0;
+    /* Same as */
+    let result = 2 + (- add 4 0);
+    /* That same example but with ppx attributes on the add application */
+    let result = 2 + (- add 4 0 [@ppx]);
+    /* Same as */
+    let result = 2 + (- add) 4 0 [@ppx];
+    /* Same as */
+    let result = 2 + (- add 4 0 [@ppx]);
+    /* Multiple nested prefixes */
+    let result = 2 + (- (- (- add 4 0)));
+    /* And with attributes */
+    let result =
+      2 + (
+        - (- (- add 4 0 [@onAddApplication]))
+      );
+
+    /**
+     * TODO: Move all of these test cases to attributes.re.
+     */
+    /* Attribute on the prefix application */
+    let res = (- something blah blah) [@attr];
```

Comment:
This is an important test case. See the difference between

`- something blah blah [@onSomethingCall]` and `(- something blah blah) [@onPrefixApplication]`.